expression/builtin_string.go

```diff
@@ -247,9 +251,17 @@ func (c *concatWSFunctionClass) getFunction(args []Expression, ctx context.Conte
 		if types.IsBinaryStr(argType) {
 			bf.tp.Flag |= mysql.BinaryFlag
```

Comment:
use `SetBinChsClnFlag(...)` to set flag/charset/collation to binary and break the execution of `for` loop